tools/coverage/gcs/gcsPostsubmit.go

```diff
@@ -95,7 +95,8 @@ func (p *PostSubmit) dirOfCompletionMarker(build int) (result string) {
 
 func (p *PostSubmit) isBuildHealthy(build int) bool {
 	return p.StorageClient.DoesObjectExist(p.Ctx, p.Bucket,
-		p.dirOfCompletionMarker(build))
+		p.dirOfCompletionMarker(build)) &&
+		p.StorageClient.DoesObjectExist(p.Ctx, p.Bucket, path.Join(p.dirOfArtifacts(build), p.covProfileName))
```

Comment:
the logic is searching backward until find a good one, there is a potential risk of comparing with something really old

Reply:
In practice that won't happen. The additional condition is false only if all unit tests fail thus no coverage profile is produced at all. When that happen, the failure should only affect post-submit jobs for a very limited time - unit test failures that affect all jobs will not stay in the master branch for long.

Reply:
Isn't this also currently true? So, what's the point of this check? Even if the baseline is not that old, you will (even for a very limited time -- how long exactly is "limited"?) get a wrong report, no?

Reply:
this is also currently true. Ideally the baseline should be the commit that the pull merge based on, anything else should be a wrong report. Currently the criteria is already lowered to rely on latest discoverable coverage report in master, and if we're ok with this logic, then it should be clear how far back we allow it go

Reply:
I checked the code and the function that lists the baseline builds does sort them from newest to oldest:

https://github.com/knative/test-infra/blob/b3c110910057090babb081ce0b1d1b1d326649c0/tools/coverage/gcs/gcsPostsubmit.go#L74

So I think we're good here.

Reply:
I think we should at least have a limit on the exhaustive for loop: https://github.com/knative/test-infra/blob/b3c110910057090babb081ce0b1d1b1d326649c0/tools/coverage/gcs/gcsPostsubmit.go#L112
An extreme case is postsubmit stopped working one day without anyone notice, and all coverage reports are wrong

Reply:
Makes sense. Can we use monitoring for catching that?

Reply:
catching postsubmit failures? I think we can, just could be noisy, as I have noticed them failed a few times for various reasons

Reply:
We should rather monitor on post-submit failures

Reply:
Instead of having a limit on the loop, I think we should fail the post-submit job with failed test and then use monitoring on it.

@chaodaiG Regarding whether to use the master branch as the baseline instead of what you proposed - it was a design decision made at the beginning of the project and the advantages were discussed before it was finalized. It was NOT of lowered criteria.

Reply:
I'm ok with the decision, my point is we should prevent it from comparing with "TOO OLD" result in master, which could be very inaccurate

Reply:
What would be the action item if monitoring found out about postsubmit failure?

Reply:
@chaodaiG if monitoring find out about postsubmit failures, oncall will be notified and they can look into  why presubmit failed multiple times

Reply:
This isn't a blocking issue for now, since the report contains absolute coverage data. We should definitely revisit this when there is need using regression ratio for PR gating

Reply:
But please add the monitoring stuff for postsubmit as we discussed

Reply:
I'll LGTM once the monitoring is in place, either here or in a different PR.

Reply:
@adrcunha Here is the PR that implements monitoring https://github.com/knative/test-infra/pull/1324

Reply:
Friendly ping. 1324 is still open, and this PR pretty dusty. :P